http_check/tests/test_http_integration.py

```diff
@@ -39,6 +40,24 @@ def test_check_cert_expiration_up(http_check):
     assert seconds_left > 0
 
 
+@pytest.mark.usefixtures("dd_environment")
+def test_cert_expiration_no_cert(http_check):
+    cert_path = os.path.join(HERE, 'fixtures', 'cacert.pem')
+    instance = {'url': 'https://valid.mock/'}
+
+    with mock.patch('ssl.SSLSocket.getpeercert', return_value={}):
+
+        status, days_left, seconds_left, msg = http_check.check_cert_expiration(instance, 10, cert_path)
+        assert status == AgentCheck.UNKNOWN
+        expected_msg = 'KeyError(\'notAfter\')'
+        if PY2:
+            expected_msg = (
```

Comment:
wow, py2 message is better?

Reply:
That was my thought as well! I'm guessing it's not always accurate